src/implementation/ncon.jl

```diff
@@ -40,10 +46,40 @@ function ncon(tensors, network,
 
     (tensors, network) = resolve_traces(tensors, network)
     tree = order === nothing ? ncontree(network) : indexordertree(network, order)
+    return ncon(tensors, network, conjlist, tree, output′; kwargs...)
+end
+function ncon(tensors, network, optimizer::T, conjlist=fill(false, length(tensors));
+              output=nothing, kwargs...) where {T<:Symbol}
+    length(tensors) == length(network) == length(conjlist) ||
+        throw(ArgumentError("number of tensors and of index lists should be the same"))
+    isnconstyle(network) || throw(ArgumentError("invalid NCON network: $network"))
+    output′ = nconoutput(network, output)
+
+    if length(tensors) == 1
+        if length(output′) == length(network[1])
+            return tensorcopy(output′, tensors[1], network[1], conjlist[1]; kwargs...)
+        else
+            return tensortrace(output′, tensors[1], network[1], conjlist[1]; kwargs...)
+        end
+    end
+
+    (tensors, network) = resolve_traces(tensors, network)
+
+    optdata = Dict{Any,Number}()
+    for (i, ids) in enumerate(network)
+        for (j, id) in enumerate(ids)
+            optdata[id] = size(tensors[i], j)
+        end
+    end
+
+    tree = optimaltree(network, optdata, TreeOptimizer{optimizer}(), false)[1]
```

Comment:
Previously I want to use that to avoid exporting the structure `TreeOptimizer`.
Now I export it together with the optimizers
```
export TreeOptimizer, ExhaustiveSearchOptimizer, GreedyMethodOptimizer, KaHyParBipartiteOptimizer, TreeSAOptimizer, SABipartiteOptimizer, ExactTreewidthOptimizer
```
so that now we can directly pass the optimizer into the function.
Do you think it's a good idea?